node/src/components/storage.rs

```diff
@@ -318,11 +320,11 @@ impl Storage {
         drop(cursor);
         block_txn.commit()?;
 
-        // Check the integrity of the block body database.
-        check_block_body_db(&env, &block_body_db)?;
+        let deleted_block_hashes_raw = deleted_block_hashes.iter().map(BlockHash::as_ref).collect();
 
-        // Check the integrity of the block metadata database.
-        check_block_metadata_db(&env, &block_metadata_db)?;
+        initialize_block_body_db(&env, &block_body_db, &deleted_block_hashes_raw)?;
+        initialize_block_metadata_db(&env, &block_metadata_db, &deleted_block_hashes_raw)?;
+        initialize_deploy_metadata_db(&env, &deploy_metadata_db, &deleted_block_hashes)?;
```

Comment:
I would separate the removal of old entries from the integrity check.

Reply:
I'd rather avoid that since we'd have to iterate each DB twice then.

Reply:
Oh right, makes sense :+1: